spark/src/test/scala/org/apache/comet/exec/CometAggregateSuite.scala

```diff
@@ -1471,6 +1471,93 @@ class CometAggregateSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("AVG and try_avg - basic functionality") {
+    withParquetTable(
+      Seq(
+        (10L, 1),
+        (20L, 1),
+        (null.asInstanceOf[Long], 1),
+        (100L, 2),
+        (200L, 2),
+        (null.asInstanceOf[Long], 3)),
+      "tbl") {
+
+      Seq(true, false).foreach({ ansiMode =>
+        // without GROUP BY
+        withSQLConf(SQLConf.ANSI_ENABLED.key -> ansiMode.toString) {
+          val res = sql("SELECT avg(_1) FROM tbl")
+          checkSparkAnswerAndOperator(res)
+        }
+
+        // with GROUP BY
+        withSQLConf(SQLConf.ANSI_ENABLED.key -> ansiMode.toString) {
+          val res = sql("SELECT _2, avg(_1) FROM tbl GROUP BY _2")
+          checkSparkAnswerAndOperator(res)
+        }
+      })
+
+      // try_avg without GROUP BY
+      val resTry = sql("SELECT try_avg(_1) FROM tbl")
+      checkSparkAnswerAndOperator(resTry)
+
+      // try_avg with GROUP BY
+      val resTryGroup = sql("SELECT _2, try_avg(_1) FROM tbl GROUP BY _2")
+      checkSparkAnswerAndOperator(resTryGroup)
+
+    }
+  }
+
+  test("AVG and try_avg - special numbers") {
+
+    val negativeNumbers: Seq[(Long, Int)] = Seq(
+      (-1L, 1),
+      (-123L, 1),
+      (-456L, 1),
+      (Long.MinValue, 1),
+      (Long.MinValue, 1),
+      (Long.MinValue, 2),
+      (Long.MinValue, 2),
+      (null.asInstanceOf[Long], 3))
+
+    val zeroSeq: Seq[(Long, Int)] =
+      Seq((0L, 1), (-0L, 1), (+0L, 2), (+0L, 2), (null.asInstanceOf[Long], 3))
+
+    val highNegNumbers: Seq[(Long, Int)] = Seq(
+      (Long.MaxValue, 1),
+      (Long.MaxValue, 1),
+      (Long.MaxValue, 2),
+      (Long.MaxValue, 2),
+      (null.asInstanceOf[Long], 3))
+
+    val inputs = Seq(negativeNumbers, highNegNumbers, zeroSeq)
+    inputs.foreach(inputSeq => {
+      withParquetTable(inputSeq, "tbl") {
+        Seq(true, false).foreach({ ansiMode =>
+          // without GROUP BY
+          withSQLConf(SQLConf.ANSI_ENABLED.key -> ansiMode.toString) {
+            val res = sql("SELECT avg(_1) FROM tbl")
+            checkSparkAnswerAndOperator(res)
+          }
+
+          // with GROUP BY
+          withSQLConf(SQLConf.ANSI_ENABLED.key -> ansiMode.toString) {
+            val res = sql("SELECT _2, avg(_1) FROM tbl GROUP BY _2")
+            checkSparkAnswerAndOperator(res)
+          }
+        })
+
+        // try_avg without GROUP BY
+        val resTry = sql("SELECT try_avg(_1) FROM tbl")
```

Comment:
maybe we can shortcut it a little bit to 

```
checkSparkAnswerAndOperator(sql("......"))
```
?

so we dont need so many names for result DF

Reply:
Sure . let me remove fetching result variable (which I used for debugging purposes)